quickwit/quickwit-cluster/src/member.rs

```diff
@@ -167,14 +158,61 @@ pub(crate) fn build_cluster_member(
         enabled_services,
         chitchat_node.gossip_public_address,
         grpc_advertise_addr,
-        running_indexing_plan,
+        indexing_tasks,
     ))
 }
 
+/// Parses indexing tasks serialized in keys formatted as `INDEXING_TASK_PREFIX:index_id:source_id`.
+/// Malformatted keys and values are ignored, just warnings are emitted.
+pub(crate) fn parse_indexing_tasks(node_state: &NodeState, node_id: &str) -> Vec<IndexingTask> {
+    node_state
+        .iter_key_values(|key, _| key.starts_with(INDEXING_TASK_PREFIX))
+        .map(|(key, versioned_value)| {
+            let (_prefix, key_without_prefix) =
+                key.split_once(INDEXING_TASK_SEPARATOR).ok_or_else(|| {
```

Comment:
There is zero chance this happens, I'd go an `expect`, which will simplify the rest of the code. There will be an issue only if we change the separator.

Reply:
Maybe I am misunderstanding something, but the guarantee that this is correct is very loose. We should not panic on that.

Reply:
I would avoid the expect... I'm always suspicious of what we can have in chitchat state....

Let me try a small refactoring to make the code looks nicer.